src/maxdiffusion/models/train.py

```diff
@@ -448,15 +448,18 @@ def compute_loss(unet_params):
         record_scalar_metrics(train_metric, new_time - last_step_completion, per_device_tflops, learning_rate_scheduler(step))
         write_metrics(writer, local_metrics_file, running_gcs_metrics, train_metric, step, config)
         last_step_completion = new_time
-        if step != 0 and (total_train_batch_size * step) % config.checkpoint_every == 0:
```

Comment:
@entrpn Hi Juan, could you check this parts?
I think the `step` refers to the index starting from 0.
And we should add one to get `num_step` to calculate ` consumed_samples`

Reply:
That makes sense!